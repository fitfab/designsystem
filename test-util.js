import React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

const themeDemo = {
  colors: {
    primary: 'blue',
    secondary: 'gray'
  }
}

const AllTheProviders = ({ children }) => {
  return (
    <ThemeProvider theme={themeDemo}>
        {children}
    </ThemeProvider>
  )
}

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

// re-export everything
export * from '@testing-library/react'

// override render method
export { customRender as render }