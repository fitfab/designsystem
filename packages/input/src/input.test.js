import React from 'react';
import { cleanup, render } from '../../../test-util'
import 'jest-styled-components';


import Input from './index';

describe('Button', () => {
    test('Render Correctly', () => {
        const { container } = render(<Input label='user name:'
        size='large'
        name= 'userName' />)
        expect(container).toMatchSnapshot()
    })
});