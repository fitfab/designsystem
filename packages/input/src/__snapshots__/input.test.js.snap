// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Correctly 1`] = `
.c1 {
  border: 1px solid #ccc;
  border-radius: 4px;
  color: #333;
  font-size: 1rem;
  font-weight: 100;
  line-height: 36px;
  padding: 0 32px;
}

.c0 {
  display: inline-block;
  margin: 0 16px 0 0;
}

<div>
  <label
    class="c0"
    for="userName"
  >
    USER NAME:
  </label>
  <input
    class="c1"
    name="userName"
  />
</div>
`;
